lib/system-task.js

```diff
@@ -3,7 +3,7 @@
 
 const fs = require("fs");
 
-module.exports = function(jobs){
+module.exports.taskPath = function taskPath(jobs){
```

Comment:
Exporting a function is a requirement of [Atom's `Task` API](https://flight-manual.atom.io/api/v1.46.0/Task/)...

Reply:
I don't think this is a requirement. `Atom's` task API is very different anyways. We don't need to follow it.

Reply:
Fixed in https://github.com/file-icons/fs/pull/8/commits/5c1f55f858343866a123c1e5a211486d050f2522

Reply:
What are you talking about? This is the file that's called by Atom's `Task` API, which [as you can see](https://github.com/atom/atom/blob/166e7d23867760b9881004143df57232c14bdf6b/src/task.coffee), takes a path to a file that's expected to export a function as its default export. Which is a seriously stupid design design, but hey, Atom are undisputed champions of bad APIs.

Reply:
I didn't see that one. I tested the code with named export and works without any issues. That is not a requirement.

Reply:
I don't know how you're testing this, but you should have seen clear breakage at startup:

<img src="https://user-images.githubusercontent.com/2346707/81372159-01f3aa00-913d-11ea-8e1c-821f5fb0d65d.png" alt="Figure 2" />

Again, could you **please** explain why you think these changes are necessary?

Reply:
This is strange. It works on my Atom. The main thing of this PR was the removal of `require.resolve`. I can bring back the default export if does not work for you.

Reply:
I repeat, **what are you trying to achieve with this pull-request**? Do you even know the difference between `require.resolve` and `require`? Or even what `require.resolve` does?

I'm so confused as to what your goal is.

Reply:
- `require.resolve` is slower.
- It doesn't work with compilers.

Reply:
> `require.resolve` is slower

\*sigh\* What the author of that [blog post](https://www.bennadel.com/blog/3243-using-require-resolve-to-calculate-module-relative-file-paths-in-node-js.htm) meant was that resolving a path *statically* is faster than resolving a path based on module resolution. Which is a given. But `require.resolve` is no slower than `require` because [it's used internally by the latter](https://nodejs.org/api/modules.html#modules_require_resolve_request_options):

> Use the internal `require()` machinery to look up the location of a module, but rather than loading the module, just return the resolved filename.

As for compilers, that's an entirely different issue, and more of a problem with whatever bundler you're using than it is with codebase structure. You haven't explained anything about what you're trying to bundle this for, so I'm at a loss as to how to help here.

Reply:
Anyways, I don't think we need to use `require.resolve` here when we clearly know what we want to `require`. Let's remove that overhead.

Reply:
Have you been paying attention to anything I've said? I already explained that Atom's `Task` API (a *class*) takes a *filesystem path to a module*, and **IT** does the `require` work. Internally. Inside Atom's core. Places I have no control or power over.

Understand now?

Reply:
What do you mean? It works now. What is the problem? I don't understand why you insist on using something with more overhead when the new suggestion works perfectly fine.

Reply:
> You just need to path a string to `Task.once`

Holy fucking shit, what do you think `require.resolve` returns?

Look. I've wasted an exorbitant amount of time trying to explain this to you. Either you don't understand, and/or you're not listening to a thing I'm saying.

Reply:
First of all, mind your tone.
Second, I am the one who is wasting his time fixing your work.

Reply:
Right, I've hit breaking point. Consider yourself banned.

Fuck off already.